### DISCUSSION TITLE
Transforming minified file leads to error

### DISCUSSION THREAD

**Discussion by JoviDeCroock** [swc-project/swc#1995] [Q&A] [answered]
I made a small reproduction, generally when calling transform with the eventual goal of minifying it leads to a Rust error:

https://codesandbox.io/s/modern-shape-3n0v3?file=/src/index.js

Code in question: https://www.runpkg.com/?preact

I know it isn't the intention to minify already minififed files but this is part of a larger bundle and it errors in the larger bundle as well

**Comment by kdy1** [comment, marked as answer]
You should do like

```js
var http = require("http");
var swc = require("@swc/core");
var fetch = require("node-fetch");

fetch("https://unpkg.com/preact")
  .then((response) => response.text())
  .then((code) => {
    swc
      .transform(code, {
        jsc: {
          target: "es2018",
          parser: {
            dynamicImport: true
          },
          minify: {
            compress: true,
            sourceMap: false,
            ecma: 2018,
            mangle: true,
            safari10: true
          }
        },
        minify: true
      })
      .then((result) => {
        console.log(result);
      })
      .catch(console.error);
  });
```

It's `jsc.minify`, not `minify`.

**Reply by kdy1** [reply]
Error message is too bad. I'll look for a way to get good error message.

**Reply by JoviDeCroock** [reply]
Thank you for the prompt response, apologies I totally missed that 😅

**Reply by JoviDeCroock** [reply]
@kdy1 here is where we're implementing it https://github.com/preactjs/wmr/pull/787#issuecomment-892014201 the difference between Terser and Swc is expected right?

**Reply by kdy1** [reply]
Yes, but only currently.


Terser has so many tests which uses `eval` and `with` which are not used in production code. I think such rules are not important and thus I prioritized rules used by real world codes.

If you point me to the input file, I will implement rules which are required for such input. So can you point them?

**Reply by kdy1** [reply]
Oh, NVM. I think I can refer to wmr repo, please tell me if it's not the case.

**Reply by JoviDeCroock** [reply]
Codebase being built https://github.com/preactjs/wmr/tree/main/examples/demo/public in wmr we transpile JSX to htm https://github.com/developit/htm maybe that is the part that doesn't compress well
